server_test.go

```diff
@@ -59,33 +59,42 @@ func TestVNCHandler(t *testing.T) {
+
+	emptyHostOptions := make([]hostOption, 0)
+	t.Run("Simple", testCase("http://example.com/vnc", 101, "localhost:5900", "localhost", 5900, false, false, emptyHostOptions, nil, false))
+	t.Run("SimpleBlockHost", testCase("http://example.com/vnc/test", 401, "", "localhost", 5900, false, false, emptyHostOptions, nil, false))
+	t.Run("SimpleBlockHostPort", testCase("http://example.com/vnc/test/1234", 401, "", "localhost", 5900, true, false, emptyHostOptions, nil, false))
+
+	t.Run("Custom", testCase("http://example.com/vnc", 101, "example.com:1234", "example.com", 1234, false, false, emptyHostOptions, nil, false))
+	t.Run("CustomHost", testCase("http://example.com/vnc/test", 101, "test:1234", "example.com", 1234, true, false, emptyHostOptions, nil, false))
+	t.Run("CustomHostPort", testCase("http://example.com/vnc/test/3456", 101, "test:3456", "example.com", 1234, true, true, emptyHostOptions, nil, false))
+
+	simpleHostOption := append(emptyHostOptions, hostOption{
+		Name: "dummy",
+		Host: "hostoption",
+		Port: "5900",
+	})
+	t.Run("SingleHostOption", testCase("http://example.com/vnc/hostoption/5900", 101, "hostoption:5900", "example.com", 1234, true, true, simpleHostOption, nil, false))
+
+	t.Run("CIDRWhitelistAllowIP", testCase("http://example.com/vnc/10.0.0.1", 101, "10.0.0.1:5900", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), true))
+	t.Run("CIDRWhitelistBlockIP", testCase("http://example.com/vnc/127.0.0.1", 401, "", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), true))
+	t.Run("CIDRBlacklistBlockIP", testCase("http://example.com/vnc/10.0.0.1", 401, "", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), false))
+	t.Run("CIDRBlacklistAllowIP", testCase("http://example.com/vnc/127.0.0.1/5900", 101, "127.0.0.1:5900", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), false))
+
+	t.Run("CIDRWhitelistAllowHost", testCase("http://example.com/vnc/10.0.0.1.ip.dns.geek1011.net", 101, "10.0.0.1.ip.dns.geek1011.net:5900", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), true))
+	t.Run("CIDRWhitelistBlockHost", testCase("http://example.com/vnc/127.0.0.1.ip.dns.geek1011.net", 401, "", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), true))
+	t.Run("CIDRBlacklistBlockHost", testCase("http://example.com/vnc/10.0.0.1.ip.dns.geek1011.net", 401, "", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), false))
+	t.Run("CIDRBlacklistAllowHost", testCase("http://example.com/vnc/127.0.0.1.ip.dns.geek1011.net/5900", 101, "127.0.0.1.ip.dns.geek1011.net:5900", "localhost", 5900, true, true, emptyHostOptions, mustParseCIDRList("192.168.0.0/24,10.0.0.0/24"), false))
```

Comment:
Can you add two more blocks of tests here:

---

- `CIDRWhitelistAllowHostOptionIP` - host option valid, resolves to ip in the whitelist, should allow
- `CIDRWhitelistBlockHostOptionIP` - host option valid, resolves to ip not in the whitelist, should block
- `CIDRBlacklistAllowHostOptionIP` - host option valid, resolves to ip not in the blacklist, should allow
- `CIDRBlacklistBlockHostOptionIP` - host option valid, resolves to ip in the blacklist, should block

---

- `CIDRWhitelistBlockBadHostOptionIP` - host option invalid, resolves to ip in the whitelist, should still block
- `CIDRWhitelistBlockBadHostOptionIP` - host option invalid, resolves to ip not in the blacklist, should block